emcc.py

```diff
@@ -1180,13 +1191,22 @@ def check(input_file):
           shared.Settings.BINARYEN_ROOT = shared.BINARYEN_ROOT
         except:
           pass
+      # default precise-f32 to on, since it works well in wasm
+      if 'PRECISE_F32=0' not in settings_changes and 'PRECISE_F32=2' not in settings_changes:
+        shared.Settings.PRECISE_F32 = 1
+      if js_opts and not force_js_opts and 'asmjs' not in shared.Settings.BINARYEN_METHOD:
+        js_opts = None
+        logging.debug('asm.js opts not forced by user or an option that depends them, and asm.js is a binaryen method, so disabling and leaving opts to the binaryen optimizer')
```

Comment:
This is a little unclear: "asm.js is a binaryen method"? wouldn't this apply to a method like interpret-binary as well?